mountpoint-s3-fs/src/metrics_otel.rs

```diff
@@ -0,0 +1,284 @@
+#[cfg(feature = "otlp_integration")]
```

Comment:
Can we conditionally export the module in the `lib` file if the feature's enabled and drop all these feature checks?